File: codegen/chains-ts/src/blockchain.ts
export interface Blockchain {
    name: string;
    description: string;
    website: string;
    explorer: string;
    logo: string;
    links: Link[];
    assets: Asset[];
}

export interface Asset {
    id: string;
    name: string;
    symbol: string;
    type: string;
    assetType: string;
    bip44CoinType: number;
    description: string;
    website: string;
    explorer: string;
    decimals: number;
    status: string;
    icon: string;
}

export interface Link {
    name: string;
    url: string;
}

File: codegen/chains-rs/src/blockchain.rs
use crate::asset::Asset;
use crate::types::Link;

pub trait Blockchain {
    fn name(&self) -> &str;
    fn description(&self) -> &str;
    fn website(&self) -> &str;
    fn explorer(&self) -> &str;
    fn links(&self) -> Vec<Link>;
    fn assets(&self) -> Vec<Box<dyn Asset>>;
    fn asset(&self, id: &str) -> Option<Box<dyn Asset>>;
    fn logo(&self) -> &str;
}
